### ISSUE TITLE
The `uuid` format is not supported ?

### ISSUE THREAD

**Issue by kripanshu** [Julian/jsonschema#737]
I am trying to validate `uuid`(v4) in the instance I provide. But it is not able to validate it properly. For example,
```
from jsonschema import validate, draft7_format_checker
schema = {
 "type" : "object",
 "properties" : {
     "price" : {"type" : "number"},
     "id" : {"type" : "string", "format":"uuid", },
 },
}
try:
  validate(instance={"id" : "hello", "price" : 34.99}, schema=schema, format_checker=draft7_format_checker)
  print('valid')
except Exception as e:
  print('Error ----> ', e)
```

expected:
```
Error -----> : Not a valid UUID
```

Result:
```
valid
```

It should have thrown error. Am I missing something in implementation?

**Comment by Julian** [comment]
There is no `uuid` format in draft 7. (You certainly can add one though.)

**Comment by gricey432** [comment]
Just updating this ticket (since it comes up in search), uuid is now a format in 2020-12 https://json-schema.org/draft/2020-12/json-schema-validation.html#rfc.section.7.3.5 so this issue should be resolved by #782

**Comment by Bktero** [comment]
I see in code that the "uuid" format is supported: https://github.com/python-jsonschema/jsonschema/blob/10fab77e2786c94560c75d24d36820c5ccd083a7/jsonschema/_format.py#L518

Is this bug related to this code?

A small test on my side shows that UUID doesn't work. The following code doesn't raise any exception:

```python
import jsonschema

schema = {
    "$schema": "https://json-schema.org/draft/2020-12/schema",
    "properties": {
        "id": {
            "type": "string",
            "format": "uuid"
        }
    }
}

jsonschema.validate({"id": "7CBD0EE5-EC30-47CE-B8B2-2DC3A363D04D"}, schema)
jsonschema.validate({"id": "7CBD0EE5-EC30-47CE-B8B2-2DC3A363D04"}, schema)  # should fail
jsonschema.validate({"id": "random text"}, schema)  # should fail
```

**Comment by Julian** [comment]
That's correct behavior, as you haven't enabled format assertion behavior.

See [the FAQ](https://python-jsonschema.readthedocs.io/en/stable/faq/#my-schema-specifies-format-validation-why-do-invalid-instances-seem-valid).

**Comment by Bktero** [comment]
My bad!

I have followed the link, installed `jsonschema[format]==4.17.3`. Still, the following code doesn't raise:

```python
from jsonschema import validate, Draft202012Validator

validate(
    instance="-12",
    schema={"format": "ipv4"},
    format_checker=Draft202012Validator.FORMAT_CHECKER,
)

validate(
    instance="-12",
    schema={"type": "string", "format": "uuid"},
    format_checker=Draft202012Validator.FORMAT_CHECKER,
)
````

The first code is from the FAQ.

Pycharm tells me "Unresolved attribute reference 'FORMAT_CHECKER' for class 'Draft202012Validator'.

In the FAQ, there is a table for the dependencies for the various format. "uuid" isn't it there, is this simply a documentation error?

**Comment by Julian** [comment]
Not sure what you mean I'm afraid, that works just fine:

```
⊙  python3.11 -m venv venv; venv/bin/python -m pip install --quiet
jsonschema'[format]==4.17.3' && venv/bin/python -c 'from jsonschema import
validate, Draft202012Validator; validate(instance="-12", schema={"format":
"uuid"}, format_checker=Draft202012Validator.FORMAT_CHECKER)'

[notice] A new release of pip available: 22.3.1 -> 23.0.1
[notice] To update, run: /Users/julian/Desktop/venv/bin/python -m pip
install --upgrade pip
Traceback (most recent call last):
  File
"/Users/julian/Desktop/venv/lib/python3.11/site-packages/jsonschema/_format.py",
line 135, in check
    result = func(instance)
             ^^^^^^^^^^^^^^
  File
"/Users/julian/Desktop/venv/lib/python3.11/site-packages/jsonschema/_format.py",
line 517, in is_uuid
    UUID(instance)
  File ***@***.***/3.11.2_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/uuid.py",
line 178, in __init__
    raise ValueError('badly formed hexadecimal UUID string')
ValueError: badly formed hexadecimal UUID string

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File
"/Users/julian/Desktop/venv/lib/python3.11/site-packages/jsonschema/validators.py",
line 1121, in validate
    raise error
jsonschema.exceptions.ValidationError: '-12' is not a 'uuid'

Failed validating 'format' in schema:
    {'format': 'uuid'}

On instance:
    '-12'
```

**Comment by Julian** [comment]
Np all good glad it works!
